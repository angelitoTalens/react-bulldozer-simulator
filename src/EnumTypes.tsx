export enum Direction {
    Up = "U",
    Down = "D",
    Left = "L",
    Right = "R",
}

export enum Commands {
    RotateLeft = "L",
    RotateRight = "R",
    Advance = "A"
}

export type Position = {
    X: number, 
    Y: number
}

export enum LandType {
    Plain = "o",
    Rocky = "r",
    TreePlanted = "t",
    Protected = "T",
    Cleared = ""
}

export enum SimulationStatus {
    Ongoing,
    NoGrid,
    OutOfBounds,
    ClearedProtected,
    UserTerminated
}

export enum TabKeys {
    Simulation = "Simulation",
    Results = "Results"
}

export type Cost = {
    quantity: number;
    value: number;
}

export type LandVisitSummary = {
    clearPlainLand: number;
    revisitClearedLand: number;
    clearRockyLand: number;
    clearTreePlantedLand: number;
}

export type Movements = {
    command: Commands;
    steps: number;
}