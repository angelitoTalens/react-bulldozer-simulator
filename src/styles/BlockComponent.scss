$block-size: 60px;
$image-size: 100%;

.block {
    background: #e7e7e7;
    border: 1px solid #999;
    float: left;
    font-size: 24px;
    font-weight: bold;
    line-height: $block-size;
    height: $block-size;
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;
    text-align: center;
    width: $block-size;
}

.step-settings-container {
    padding-top: 20px;
    padding-left: 7px;
    padding-right: 5px;
}

img {
    max-width: $image-size;
    max-height: $image-size;
}