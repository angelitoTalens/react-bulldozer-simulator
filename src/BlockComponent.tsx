import React from "react";
import "./styles/BlockComponent.scss"
import { LandType } from "./EnumTypes"
import Col from "react-bootstrap/Col";
import { Direction } from "./EnumTypes";

import bulldozer from "./images/bulldozer.gif"


type BlockComponentProps = {
    landType: LandType;
    currentPos: boolean;
    direction: Direction;
}


function BulldozerDirection(direction: Direction) {
    let style = {transform: "rotateY(180deg)"};
    switch(direction) {
        case Direction.Left:
            style = {transform: "none"}
            break;
        case Direction.Up:
            style = {transform: "rotateY(180deg) rotateZ(90deg)"};
            break;
        case Direction.Down:
            style = {transform: "rotateY(180deg) rotateZ(270deg)"};
            break;
        default:
            break; 
    }
    return style;
}

function BlockComponent(props: BlockComponentProps) {
    
    if (props.currentPos) {
        return (
            <Col className="block current-block">
                <img src={bulldozer} alt="bulldozer" style={BulldozerDirection(props.direction)}/>
            </Col>
        );
    }
    else {
        return (
            <Col className="block">
                {props.landType}
            </Col>
        );
    }
}

export default BlockComponent;
