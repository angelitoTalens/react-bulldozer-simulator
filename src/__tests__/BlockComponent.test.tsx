import React from 'react';
import renderer from 'react-test-renderer';

import BlockComponent from '../BlockComponent';
import { LandType, Direction } from '../EnumTypes';

test ("Test BlockComponent", ()  => {

    const directions: Direction[] = [Direction.Down, Direction.Left, Direction.Up, Direction.Right] 
    for (let d of directions) {
        const component = renderer.create(
            <BlockComponent landType={LandType.Plain} currentPos={true} direction={d}/>
        );
    
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    }
});