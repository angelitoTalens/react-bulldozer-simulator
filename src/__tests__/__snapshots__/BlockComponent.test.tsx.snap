// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BlockComponent 1`] = `
<div
  className="block current-block col"
>
  <img
    alt="bulldozer"
    src="bulldozer.gif"
    style={
      Object {
        "transform": "rotateY(180deg) rotateZ(270deg)",
      }
    }
  />
</div>
`;

exports[`Test BlockComponent 2`] = `
<div
  className="block current-block col"
>
  <img
    alt="bulldozer"
    src="bulldozer.gif"
    style={
      Object {
        "transform": "none",
      }
    }
  />
</div>
`;

exports[`Test BlockComponent 3`] = `
<div
  className="block current-block col"
>
  <img
    alt="bulldozer"
    src="bulldozer.gif"
    style={
      Object {
        "transform": "rotateY(180deg) rotateZ(90deg)",
      }
    }
  />
</div>
`;

exports[`Test BlockComponent 4`] = `
<div
  className="block current-block col"
>
  <img
    alt="bulldozer"
    src="bulldozer.gif"
    style={
      Object {
        "transform": "rotateY(180deg)",
      }
    }
  />
</div>
`;
